operators/pkg/controller/elasticsearch/name/name.go

```diff
@@ -62,13 +64,28 @@ func suffix(name string, sfx string) string {
 
 // NewPodName returns a unique name to be used for the pod name and the
 // Elasticsearch cluster node name.
-func NewPodName(esName string) string {
-	sfx := stringsutil.Concat(
-		string(podSuffix),
-		"-",
-		rand.String(podRandomSuffixLength),
-	)
-	return suffix(esName, sfx)
+func NewPodName(esName string, nodeSpec v1alpha1.NodeSpec) string {
```

Comment:
I think this function has become complex enough so it would be nice to have an additional comment like:
```go
The pod name will respect the pattern "<$clusterName>-es-[$nodeName-]<random suffix>".
```